framework/decode/dx12_resource_value_mapper.cpp

```diff
@@ -1699,10 +1731,89 @@ void Dx12ResourceValueMapper::GetStateObjectLrsAssociationInfo(
             auto num_exports           = dxil_lib_desc_decoder->GetPointer()->NumExports;
             if (num_exports == 0)
             {
-                // TODO: Parse the names of all shaders exported from the DXIL library.
+#if defined(GFXRECON_DXC_SUPPORT)
+                // If D3D12_DXIL_LIBRARY_DESC::NumExports == 0, everything in the DXIL library is exported. Use
+                // reflection to get the list of exported shader names.
+                HRESULT                         hr;
+                graphics::dx12::IDxcUtilsComPtr dxc_utils = nullptr;
+                hr                                        = DxcCreateInstance(CLSID_DxcUtils, IID_PPV_ARGS(&dxc_utils));
+                if (SUCCEEDED(hr))
+                {
+                    graphics::dx12::IDxcContainerReflectionComPtr dxc_container_reflection = nullptr;
+                    DxcCreateInstance(CLSID_DxcContainerReflection, IID_PPV_ARGS(&dxc_container_reflection));
+                    if (SUCCEEDED(hr))
+                    {
+                        // Create a DXC blob from the DXIL library's bytes.
+                        graphics::dx12::IDxcBlobEncodingComPtr dxc_blob_encoding;
+                        hr = dxc_utils->CreateBlob(dxil_lib_desc_decoder->GetPointer()->DXILLibrary.pShaderBytecode,
```

Comment:
I think it should be possible to use CreateBlobFromPinned here to avoid a copy to a new allocation,

Reply:
Changed `CreateBlob` to `CreateBlobFromPinned`.